src/main/java/club/sk1er/hytilities/handlers/chat/modules/modifiers/DefaultChatRestyler.java

```diff
@@ -168,4 +169,25 @@ public void onMessageReceived(@NotNull ClientChatReceivedEvent event) {
 
         Hytilities.INSTANCE.getChatHandler().fixStyling(event.message, siblings);
     }
+
+    /**
+     * Handles the replacement of channel names
+     * Loops through all siblings to find a replacement
+     *
+     * @param message The message being modified
+     * @param pattern The regular expression to check the message and it's components against
+     * @param replacement The text that replaces what is matched by the regular expression
+     * @param checkParentComponent Whether or not to check the parent chat component
+     */
+    private ChatComponentText shortenChannelName(IChatComponent message, String pattern, String replacement, boolean checkParentComponent) {
+        String originalText = message.getUnformattedTextForChat();
+        if (checkParentComponent && !originalText.contains("\u00a7")) {
+            originalText = (message.getChatStyle().getFormattingCode() + originalText + EnumChatFormatting.RESET).replaceAll(pattern, replacement);
+        }
+        ChatComponentText copy = (ChatComponentText) new ChatComponentText(originalText).setChatStyle(message.getChatStyle());
+        for (IChatComponent sibling : message.getSiblings()) {
+            copy.appendSibling(new ChatComponentText(sibling.getUnformattedTextForChat().replaceAll(pattern, replacement)).setChatStyle(sibling.getChatStyle()));
```

Comment:
Instead of using replaceAll, it may be possible to replace only one time since there is only one possible occurrence in an entire message?